cli/tests/pcluster/templates/test_shared_storage.py

```diff
@@ -202,6 +203,21 @@ def assert_sg_rule(
     assert_that(sg_rules).is_length(1)
 
 
+def test_non_happy_ontap_and_openzfs_mounting(mocker, test_datadir):
+    dummy_api = _DummyAWSApi()
+    dummy_api._fsx.set_non_happy_describe_volumes(True)
```

Comment:
set_non_happy_describe_volumes(True) says that we are injecting a failure in the describe volumes API call, but it does not make any specification about the nature of such failure. Internally you’re injecting a client error saying unauhtorized exception, but you’re not in control of that

So, we have two options:
rename set_non_happy_describe_volumes  in a way that reflects the fact that you’re forcing an unauthorized error
add a parameter to set_non_happy_describe_volumes  to specify the nature of the raised error

Reply:
Ok, this is addressed here [add error parameter for set_non_happy_describe_volumes](https://github.com/aws/aws-parallelcluster/pull/4749/commits/784f24a26faefe0a7fa76fab0e2e498aa1ca71b5) and [remove boolean parameter from set_non_happy_describe_volumes](https://github.com/aws/aws-parallelcluster/pull/4749/commits/d7da791e830d9555bbb9ce52d84174302efead58)